src/Datagrid/PagerInterface.php

```diff
@@ -93,12 +94,24 @@ public function setQuery($query);
 //    public function haveToPaginate(): bool;
 
     /**
+     * NEXT_MAJOR: Remove this method.
+     *
+     * @deprecated since sonata-project/admin-bundle 3.x. To be removed in 4.0. Use getCurrentPageResults() instead.
+     *
      * Returns an array of results on the given page.
      *
-     * @return object[]
+     * @return iterable<object>
      */
     public function getResults();
 
+//    NEXT_MAJOR: uncomment this method in 4.0
+//    /**
+//     * Returns an array of results on the given page.
```

Comment:
Since the return type is declared as `iterable`, I guess we should use "collection" instead of "array" in the description:
```suggestion
//     * Returns a collection of results on the given page.
```

Reply:
Done